src/ray/pubsub/publisher.cc

```diff
@@ -90,24 +101,32 @@ const absl::flat_hash_map<SubscriberID, SubscriberState *> &EntityState::Subscri
   return subscribers_;
 }
 
+SubscriptionIndex::SubscriptionIndex(rpc::ChannelType channel_type)
```

Comment:
nit: I feel it's a little bit wired to put channel_type into SubscriptionIndex. it seems that it's only used to construct EntityState. We don't need to store it inside `channel_type_`.

Reply:
Another thing I feel bad about is that it's an application layer decision, but here it's hardcoded in the infra layer.  I'm wondering whether we can make it better?

Reply:
synced offline. I'm good with this right now since it's still maintainable.